File: src/client/SimpleAppClient/src/app/product/models/product.ts
export class Product {
    productId?: number;
    categoryId:number;
    productName:string;
    description:string;
    unitPrice:number;

    constructor(){
        this.productId = 0;
        this.categoryId = 0;
        this.productName = null;
        this.description = null;
        this.unitPrice = 0;
    }
}

File: src/server/SimpleAppBackend/SimpleApp.Domain/Entities/Product.cs
using SimpleApp.Domain.Common;
using System;
using System.Collections.Generic;
using System.Reflection;
using System.Text;

namespace SimpleApp.Domain.Entities
{
    public class Product : AuditableEntity
    {
        public int ProductId { get; set; }
        public string ProductName { get; set; }
        public string Description { get; set; }
        // decimal unit price will be problematic in real situation for fractions
        public decimal UnitPrice { get; set; }

    }
}
